### PR TITLE
stm32 development page

### PR DESCRIPTION
**Pull request by olliw42**
am wondering if we shouldn't clean the main project page and move stm32 dev talk to here

### CODE REVIEW

File: docs/STM32_DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+([back to main page](../README.md))
+
+mLRS uses STM32CubeIDE for STM32 targets. For STM32 targets this procedure should work:
+
+## Software: Installation Bits and Bops ##
+
+Let's assume that the project should be located in the folder C:/Me/Documents/Github/mlrs.
+ 
+### I. Clone and setup the project files ###
```
Comment (jlpoltrack) [inline docs/STM32_DEVELOPMENT.md:11]: Would think the Roman numerals aren't needed.

File: docs/STM32_DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+8. compiling should work now: Go to the green 'right-pointing triangle' icon in the top icon bar and click it
+9. repeat steps 4. - 8. for each board you are interested in
+
+<img src="https://user-images.githubusercontent.com/6089567/154903396-25f62bf6-573a-4b80-9720-a0ad4a21f291.jpg" width="480">
+
+The STM32CubeIDE has its weirdness, so you may have to get used to it. 
+
+In case of issues with this procedure, don't hesitate to join the discussion thread at rcgroups or the discord channel, or submit an issue in the github repository.
+
+#### Dependencies ####
```
Comment (jlpoltrack) [inline docs/STM32_DEVELOPMENT.md:43]: Think this would be better suited to before the installation steps.

File: docs/STM32_DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+Let's assume that the project should be located in the folder C:/Me/Documents/Github/mlrs.
+ 
+### I. Clone and setup the project files ###
+
+1. open a command line processor
+2. cd into `C:/Me/Documents/Github` (not C:/Me/Documents/Github/mlrs !)
+3. `git clone https://github.com/olliw42/mLRS.git mlrs`
+4. `cd mlrs`
+5. run `run_setup.py`. This does three steps: Initializes submodules (git submodule --init --recursive), copies ST HAL and LL drivers to the target folders, and generates the MAVLink library files.
```
Comment (jlpoltrack) [inline docs/STM32_DEVELOPMENT.md:17]: Maybe subbullets here.

File: docs/STM32_DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - ***Note***: Ensure that all three steps are executed completely.
+
+For cloning you of course can use any other tool you like.
+
+### II. STM32CubeIDE (for STM32 targets) ###
```
Comment (jlpoltrack) [inline docs/STM32_DEVELOPMENT.md:22]: Same note about Roman numerals.

**Comment by jlpoltrack** [comment]
> am wondering if we shouldn't clean the main project page and move stm32 dev talk to here

Not a bad idea - the ReadMe is rather long and this part is right at the bottom.

Some comments - nothing too major.

**Comment by olliw42** [comment]
many thx for the comments. MAde soem changes. I like to snot use the subbullets since the numbers make it muich easier to refer to some particular steps. For teh same rason I kept the roman numbers in teh titles, but moved them to the back.

File: docs/STM32_DEVELOPMENT.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# mLRS Documentation: Building and Flashing for STM32 Devices #
+
+([back to main page](../README.md))
+
+mLRS uses STM32CubeIDE for STM32 targets. The following procedure should work.
```
Comment (jlpoltrack) [inline docs/STM32_DEVELOPMENT.md:5]: Nitpick - the title says '...for STM32 Devices' but here says '...for STM32 targets' don't really prefer one or other but think that they should be consistent.

**Comment by jlpoltrack** [comment]
Sounds good - just one more comment and then will merge.

File: docs/STM32_DEVELOPMENT.md (hunk 5 above)
Comment (olliw42) [reply docs/STM32_DEVELOPMENT.md:5]: hm ... interesting. It's btw so also for the ESP page.
I see your point, but somehow I dont like targets in the title. It's really no argument, but "devices" sounds more like the actual thing and also a bit more general to me.
maybe we could change the first sentence in some way

Comment (jlpoltrack) [reply docs/STM32_DEVELOPMENT.md:5]: Let's not overthink then - will merge.

**Pull request merged**